miio/airconditioner_miot.py

```diff
@@ -0,0 +1,403 @@
+    # Electricity (siid=8)
+    "electricity": {"siid": 8, "piid": 1},
+    # Maintenance (siid=9)
+    "clean": {"siid": 9, "piid": 1},
+    "running_duration": {"siid": 9, "piid": 5},
+    # Enhance (siid=10)
+    "fan_percent": {"siid": 10, "piid": 1},
+    "timer": {"siid": 10, "piid": 3},
+}
+
+
+class AirConditionerMiotException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    Cool = 2
+    Dry = 3
+    Fan = 4
+    Heat = 5
+
+
+class FanLevel(enum.Enum):
+    Auto = 0
+    Level1 = 1
+    Level2 = 2
+    Level3 = 3
+    Level4 = 4
+    Level5 = 5
+    Level6 = 6
+    Level7 = 7
+
+
+class AirConditionerMiotStatus:
+    """Container for status reports from the air conditioner which uses MIoT protocol."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def is_on(self) -> bool:
+        """True if the device is turned on."""
+        return self.data["power"]
+
+    @property
+    def power(self) -> str:
+        """Current power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["mode"])
+
+    @property
+    def target_temperature(self) -> float:
+        """Target temperature in Celsius."""
+        return self.data["target_temperature"]
+
+    @property
+    def eco(self) -> bool:
+        """True if ECO mode is on."""
+        return self.data["eco"]
+
+    @property
+    def heater(self) -> bool:
+        """True if aux heat mode is on."""
+        return self.data["heater"]
+
+    @property
+    def dryer(self) -> bool:
+        """True if aux dryer mode is on."""
+        return self.data["dryer"]
+
+    @property
+    def sleep_mode(self) -> bool:
+        """True if sleep mode is on."""
+        return self.data["sleep_mode"]
+
+    @property
+    def fan_level(self) -> FanLevel:
+        """Current Fan level."""
+        return FanLevel(self.data["fan_level"])
+
+    @property
+    def vertical_swing(self) -> bool:
+        """True if vertical swing is on."""
+        return self.data["vertical_swing"]
+
+    @property
+    def temperature(self) -> float:
+        """Current ambient temperature in Celsius."""
+        return self.data["temperature"]
+
+    @property
+    def buzzer(self) -> bool:
+        """True if buzzer is on."""
+        return self.data["buzzer"]
+
+    @property
+    def led(self) -> bool:
+        """True if LED is on."""
+        return self.data["led"]
+
+    @property
+    def electricity(self) -> float:
+        """Power consumption accumulation in kWh."""
+        return self.data["electricity"]
+
+    @property
+    def clean(self) -> str:
+        """Auto clean mode indicator."""
+        return self.data["clean"]
+
+    @property
+    def running_duration(self) -> float:
+        """Total running duration in hours."""
+        return self.data["running_duration"]
```

Comment:
Would it make sense to return a timedelta here?

Reply:
This property seems to return total running time since its first power on, maybe it's a statistical data for manufacture to refer. Even I unplug and re-plug it back doesn't reset this value.

Reply:
Maybe the property should then be renamed to `total_running_time` or alike? In any case, I think this should be returned as `timedelta`.